contentcuration/contentcuration/views.py

```diff
@@ -270,6 +276,67 @@ def _duplicate_node(node, sort_order=1, parent=None):
 
     return new_node
 
+
+def move_nodes(request):
+    logging.debug("Entering the move_nodes endpoint")
+
+    if request.method != 'POST':
+        raise HttpResponseBadRequest("Only POST requests are allowed on this endpoint.")
+    else:
+        data = json.loads(request.body)
+
+        try:
+            node_ids = data["node_ids"]
+            target_parent = data["target_parent"]
+            channel_id = data["channel_id"]
+        except KeyError:
+            raise ObjectDoesNotExist("Missing attribute from data: {}".format(data))
+
+        logging.info("Moving node id %s", node_ids)
+
+        nodes = [json.loads(n) for n in node_ids.split()]
+        nodes_list = []
+
+        with transaction.atomic():
+            for node_data in nodes:
+                nodes_list.append(_move_node(node_data['id'], sort_order=node_data['sort_order'], parent=target_parent, channel_id=channel_id))
+
+        nodes_serializer = ContentNodeSerializer(nodes_list, many=True)
+        return HttpResponse(json.dumps({
+            "success": True,
+            "nodes": JSONRenderer().render(nodes_serializer.data)
+        }))
+
+def _move_node(node, sort_order=1, parent=None, channel_id=None):
+    if isinstance(node, int) or isinstance(node, basestring):
+        node = ContentNode.objects.get(pk=node)
+
+    node.parent = ContentNode.objects.get(pk=parent) if parent else None
+    node.changed = True
+    node.sort_order = sort_order
+    descendants = node.get_descendants(include_self=True)
+    to_save = [node]
+    for tag in ContentTag.objects.filter(tagged_content__in=descendants):
+        # If moving from another channel
+        if tag.channel_id != channel_id:
+            t, is_new = ContentTag.objects.get_or_create(
+                tag_name=tag.tag_name,
+                channel_id=channel_id,
+            )
+
+            # Set descendants with this tag to correct tag
+            for n in descendants.filter(tags=tag):
+                if n in descendants:
+                    n.tags.remove(tag)
+                    n.tags.add(t)
+                    to_save.append(n)
+
+    # Save all updates to nodes with changed tags
+    for n in to_save:
+        n.save()
```

Comment:
(So, we just need to save the top-level `node` itself, but that can happen above, before the loop).